### PR TITLE
docs: add forms terms to glossary

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/glossary.md [hunk 1]
```diff
@@ -577,12 +603,12 @@ The alternative is a [template-driven form](guide/glossary#template-driven-forms
 
 When building reactive forms:
 
-* The "source of truth" is the component. The validation is defined using code in the component.
-* Each control is explicitly created in the component class with `new FormControl()` or with `FormBuilder`.
+* The "source of truth", the form model, is defined in the component class. The validation is defined using code in the component.
+* Each control is explicitly created in the component class by creating a `FormControl` instance manually or with `FormBuilder`.
```
Comment (jbogarthyde) [inline aio/content/guide/glossary.md]: Instead of "The validation is defined...", perhaps "Validation functions are defined in the component."

**Review by jbogarthyde — COMMENTED** [review]
Some editorial suggestions.

File: aio/content/guide/glossary.md [hunk 2]
```diff
@@ -317,6 +317,24 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
...
+
+One of the three fundamental building blocks of Angular forms, along with `FormGroup` and `FormArray` to track the value,
+validation, and status of a form input [element](guide/glossary#element).
+
+{@a form-model}
+
+## form model
+
+The "source of truth" for a form element at a given point in time. With [reactive forms](guide/glossary#reactive-forms), the form model is created explicitly in the component class. With [template-driven forms], the form model is implicitly created by directives.
+
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: ...plus, add link to template-driven forms term

File: aio/content/guide/glossary.md [hunk 3]
```diff
@@ -325,6 +343,14 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
...
 {@a I}
 
+
+{@a immutability}
+
+## immutability
+
+The ability to alter the state of a value after its creation. [Reactive forms](guide/glossary#reactive-forms) promote immutability in that
+each change to the data model produces a new data model rather than modifying the existing one. [Template-driven forms](guide/glossary#template-driven-forms) promote mutable changes with `NgModel` and [two-way data binding](guide/glossary#data-binding) to modify the existing data model in-place.
+
```
Comment (jbogarthyde) [inline aio/content/guide/glossary.md]: "in place" doesn't have a hyphen here - only when it is used as an adjective - "in-place modification".

**Review by kara — CHANGES_REQUESTED** [review]

File: aio/content/guide/glossary.md [hunk 4]
```diff
@@ -317,6 +317,24 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
 
 {@a F}
 
+{@a form-control}
+
+## form control
+
+A fundamental building block for Angular forms. The `FormControl`, together with `FormGroup` and `FormArray`, tracks the value, validation, and status of a form input [element](guide/glossary#element).
```
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: It's a bit unusual to define a term (form control) and then immediately change the term (the FormControl). How about this: A instance of `FormControl`, which is a fundamental building block for Angular forms. Together with `FormGroup` and `FormArray`, tracks the value, validation, and status of a form input [element](guide/glossary#element).

File: aio/content/guide/glossary.md [hunk 5]
```diff
@@ -317,6 +317,24 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
 
 {@a F}
 
+{@a form-control}
+
+## form control
```
[the remark this answers is not in the document]
Comment (jenniferfell) [reply aio/content/guide/glossary.md]: Fwiw: In my doc comments, I recommend this style: "A form control" and "An instance of `FormControl'" but not "a `FormControl`".  I think it's good to define a concept like "form control" as a regular noun (similar to component, template, etc) and then as its definition say that it's an instance of the class `FormControl.`

File: aio/content/guide/glossary.md (hunk 4 above)
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: Is this the same definition of "element" as what we have in the Glossary for Angular element? Feels like this is just a standard HTML element? If so, probably best to remove link to glossary def for Angular element.

Comment (jenniferfell) [inline aio/content/guide/glossary.md]: Recommendation: Add link to Introduction to Forms

File: aio/content/guide/glossary.md [hunk 6]
```diff
@@ -317,6 +317,24 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
...
+
+A fundamental building block for Angular forms. The `FormControl`, together with `FormGroup` and `FormArray`, tracks the value, validation, and status of a form input [element](guide/glossary#element).
+
+{@a form-model}
+
+## form model
+
+The "source of truth" for the value and validation status of a form input element at a given point in time. With [reactive forms](guide/glossary#reactive-forms), the form model is created explicitly in the component class. With [template-driven forms](guide/glossary#template-driven-forms), the form model is implicitly created by directives.
```
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: A nit: "With" sounds like companionship. Maybe "When using reactive forms" or "When used with reactive forms"?

Comment (jenniferfell) [inline aio/content/guide/glossary.md]: Recommendation: Add link to Introduction to Forms

File: aio/content/guide/glossary.md [hunk 7]
```diff
@@ -317,6 +317,24 @@ Within Angular, use [NgModules](guide/glossary#ngmodule) to make public parts av
...
+
+{@a form-validation}
+
+## form validation
+
+A check that runs when form values change and reports whether the given values are correct and complete, according to the defined constraints. Reactive forms apply [validator functions](guide/form-validation#adding-to-reactive-forms). Template-driven forms use [validator directives](guide/form-validation#adding-to-template-driven-forms).
```
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: Nice! Recommendation: Link to forms validation doc?

File: aio/content/guide/glossary.md [hunk 8]
```diff
@@ -577,12 +603,13 @@ The alternative is a [template-driven form](guide/glossary#template-driven-forms
 
 When building reactive forms:
 
-* The "source of truth" is the component. The validation is defined using code in the component.
-* Each control is explicitly created in the component class with `new FormControl()` or with `FormBuilder`.
+* The "source of truth", the form model, is defined in the component class.
```
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: Let's be precise with the term "build". Recommend "using" or "implementing" here.

File: aio/content/guide/glossary.md [hunk 9]
```diff
@@ -577,12 +603,13 @@ The alternative is a [template-driven form](guide/glossary#template-driven-forms
...
+* Validation is set up through validation functions rather than valdation directives.
+* Each control is explicitly created in the component class by creating a `FormControl` instance manually or with `FormBuilder`.
 * The template input elements do *not* use `ngModel`.
-* The associated Angular directives are prefixed with `Form`, such as `FormGroup()`, `FormControl()`, and `FormControlName()`.
+* The associated Angular directives are prefixed with `form`, such as `formControl`, `formGroup`, and `formControlName`.
 
-Reactive forms are powerful, flexible, and a good choice for more complex data-entry form scenarios, such as dynamic generation of form controls.
+Reactive forms are powerful, flexible, and more scalable for complex data-entry form scenarios, such as dynamic generation of form controls.
```
Comment (jenniferfell) [inline aio/content/guide/glossary.md]: I recommend deleting this. It's a bit sales-pitchy. I would replace it with the sentence from above a link to the Intro to Forms for a comparison: The alternative is a [template-driven form](guide/glossary#template-driven-forms). For an introduction and comparison of both forms approaches, see Introduction to Angular Forms.

Comment (jenniferfell) [reply aio/content/guide/glossary.md]: P.S. Same comments on the definition for template-driven forms.

**Review by jenniferfell — CHANGES_REQUESTED** [review]

File: aio/content/guide/glossary.md (hunk 4 above)
Comment (kara) [reply aio/content/guide/glossary.md]: Much better, I like that.

**Review by kara — APPROVED** [review]
LGTM (after @jenniferfell's comments are addressed)

**Review by jenniferfell — APPROVED** [review]

**Pull request closed without merging**